controllers/autohealconfig_controller_test.go

```diff
@@ -0,0 +1,693 @@
+		err    bool
+	}{
+		{
+			name: "Test secret is present and struct gets populated",
+			fields: fields{
+				fake.NewClientBuilder().WithRuntimeObjects(&core.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: autoheal.Namespace, Name: autoheal.Spec.SecretName},
+					Data: map[string][]byte{
+						"tag":          []byte(tag),
+						"access_token": []byte(accessToken),
+						"repository":   []byte(repoUrl),
+					},
+				}).Build(),
+			},
+			args: args{
+				ctx:      context.TODO(),
+				autoheal: &autoheal,
+				status:   status,
+			},
+			want: &v1alpha1.ConfigRepositoryInfo{
+				Tag:           tag,
+				AccessToken:   accessToken,
+				RepositoryURL: repoUrl,
+			},
+			err: false,
+		},
+		{
+			name: "Test secret is not present and function returns nothing",
+			fields: fields{
+				fake.NewClientBuilder().WithRuntimeObjects(&core.Secret{}).Build(),
+			},
+			args: args{
+				ctx:      context.TODO(),
+				autoheal: &autoheal,
+				status:   status,
+			},
+			want: nil,
+			err:  false,
+		},
+		{
+			name: "Test secret is present but is missing values function returns nothing",
+			fields: fields{
+				fake.NewClientBuilder().WithRuntimeObjects(&core.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: autoheal.Namespace, Name: autoheal.Spec.SecretName},
+					Data: map[string][]byte{
+						"tag":          []byte(""),
+						"access_token": []byte(accessToken),
+						"repository":   []byte(repoUrl),
+					},
+				}).Build(),
+			},
+			args: args{
+				ctx:      context.TODO(),
+				autoheal: &autoheal,
+				status:   status,
+			},
+			want: nil,
+			err:  false,
+		},
+		{
+			name: "Test secret is present but url cannot be parsed returns nothing, error thrown",
+			fields: fields{
+				fake.NewClientBuilder().WithRuntimeObjects(&core.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: autoheal.Namespace, Name: autoheal.Spec.SecretName},
+					Data: map[string][]byte{
+						"tag":          []byte(tag),
+						"access_token": []byte(accessToken),
+						"repository":   []byte("bad repo url"),
+					},
+				}).Build(),
+			},
+			args: args{
+				ctx:      context.TODO(),
+				autoheal: &autoheal,
+				status:   status,
+			},
+			want: nil,
+			err:  true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := AutohealConfigReconciler{
+				Client: tt.fields.Client,
+			}
+			repoSecret, err := r.getConfigRepoSecret(tt.args.ctx, tt.args.autoheal, tt.args.status)
+			if (err != nil) != tt.err {
+				t.Errorf("GetConfigRepoSecret() error = %v wantedError= %v", err, tt.err)
+
+			}
+			if !reflect.DeepEqual(repoSecret, tt.want) {
+				t.Errorf("GetConfigRepoSecret() got = %v, want %v", repoSecret, tt.want)
+			}
+		})
+	}
+
+}
+
+func TestReadFiles(t *testing.T) {
+
+	type fields struct {
+		httpClient *httptest.Server
+	}
+
+	type args struct {
+		repo *v1alpha1.ConfigRepositoryInfo
+	}
+	mockGood := mockHttpClientGood(t)
+	mockBad := mockHttpClientBad(t)
+	mockGoodBad := mockHttpClientGoodBad(t)
```

Comment:
```suggestion
	mockGood := mockHttpClientGood()
	mockBad := mockHttpClientBad()
	mockGoodBad := mockHttpClientGoodBad()
```
Looks like we are passing in the `t` but the function doesn't use it, so it can be removed from here and also the reference in the function.

Reply:
Yes good catch copy and paste error